content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,562 @@
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        System.out.println(cart);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart.removeProductsFromCartByAdmin(1L,product);
+        System.out.println(cart);
+        assertEquals(2,cart.totalNumberOfItems());
+
+
+    }
+    @Test
+    void checkAllTheProductsInTheCartByCustomer(){
+        cart.refresh();
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Product product3 = new Product("3", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product3);
+        System.out.println(  cart.checkAllTheProductsInTheCartByCustomer(1L));
+    }
+    @Test
+    void checkAllRegisteredAdmin(){
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        System.out.println(dataService.getAllAdmin());
+    }
+}
+```
+### Using Postman to verify the output on the web
+First things first go to learning.postman.com and download the version of postman that suits your operating system. You will have to do the necessary sign-in if it's your first time. You will have to create a new environment by clicking under the `Scratch Pad`. On the three (3) dots in front of the environment, click on it to rename the environment and start operation. You click on the plus or add button in front of your environment and then a new interface will appear on your screen. Postman provides those HTTP methods that we already have in our controller that help us perform operations on request, Examples are the POST, PUT, PATCH, GET, DELETE, etc. Now you are making these operations from your local machine, so you start with localhost: port-number, followed by the string on your controller, and the string in the individual method, eg localhost:8090/cart/create_customer
+
+![shopping_api](api.JPG)
+
+![shopping_api1](api2.JPG)
+
+![shopping_api2](payment.JPG)
+### Conclusion
+In the end, we've seen how to code to an interface, how to write a unit test, how to create an endpoint, and how to use postman to view our endpoint. Also, we discussed how we used field dependency injection and not constructor dependency injection which will still suffice. Furthermore, we also added products to our cart and the cart could calculate the cost of products in the cart since the products have an attribute of price. 
```

Comment:
This talks about how you discussed field dependency injection vs constructor dependency injection yet you have never talked about it earlier. Please either remove the sentence: 

> Also, we discussed how we used field dependency injection and not constructor dependency injection which will still suffice.

Or add a brief section talking about field dependency injection vs constructor dependency injection earlier in the article.